### ISSUE TITLE
Data filtering doesn't work in transform_lookup?

### ISSUE THREAD

**Issue by MariaKokshaikina** [vega/altair#2309]
I would like to build an interactive chart with world map with COVID-19 confirmed cases per country and date slider using Altair library in python.

Data format:
```
"country_region","date","confirmed_cases"
"Afghanistan",2020-01-22,0
"Afghanistan",2020-01-23,0
"Afghanistan",2020-01-24,0
"Afghanistan",2020-01-25,0
```

I have managed to build non-interactive chart, but can't set up slider. My chart now looks like this: 
![visualization](https://user-images.githubusercontent.com/10485273/95268638-66c44500-0840-11eb-9081-3009deed0f51.png)

I want to add slider to chose date, but can't set up it properly.

I use `transform_lookup` method to add colors to countries according to the data (confirmed cases / population) and `transform_filter` as far as I understand should be added to perform data filtering when slider position is changed by user. But this doesn't work. **My assumption is that in `transform_lookup` always original data is used and filtering doesn't work there despite I've passed `transform_filter` right before it**. I haven't found examples or documentation about using `transform_lookup` and slider at the same time. If I am wrong, I would be very grateful to hear any ideas what exactly I do wrong and what can help me with this problem.

Code:
```
import requests
import json

import pandas as pd

import altair as alt
from vega_datasets import data
from altair import datum

df = pd.read_csv('https://raw.githubusercontent.com/MariaKokshaikina/any-data/main/covid19_global_confirmed_cases%20(1).csv')

country_info = requests.get(
    'https://raw.githubusercontent.com/MariaKokshaikina/any-data/main/country_info.json'
).json()

df = df[df['country_region'].isin(country_info)]
df = df.sort_values('date', ascending=True)
df = df.tail(5000)

def timestamp(t):
    return pd.to_datetime(t).timestamp() * 1000

df['id'] = df['country_region'].map(lambda x: country_info[x]['numericCode'])
df['rate'] = df['confirmed_cases'] / df['country_region'].map(lambda x: country_info[x]['population'])
df['timestamp'] = df['date'].map(timestamp)

countries = alt.topo_feature(data.world_110m.url, 'countries')

slider = alt.binding_range(
    step=24 * 60 * 60 * 1000,
    min=df['timestamp'].min(), 
    max=df['timestamp'].max()
)

select_date = alt.selection_single(
    name="slider", 
    fields=['timestamp'],
    bind=slider, 
)

alt.Chart(countries).mark_geoshape()\
    .encode(color='rate:Q')\
    .add_selection(select_date)\
    .transform_filter(select_date)\
    .transform_lookup(
        lookup='id',
        from_=alt.LookupData(df, key='id', fields=['rate'])
    )\
    .project('equirectangular')\
    .properties(
        width=500,
        height=300,
        title='Title'
    )

```

* You can remove `.transform_filter(select_date)\` line to see working map with slider which doesn't change anything.

Thank you!

**Comment by jakevdp** [comment]
Answered on StackOverflow: https://stackoverflow.com/questions/64234865/world-map-slider-in-altair-python/64235215

**Comment by joelostblom** [comment]
@MariaKokshaikina I am going through Altair issues to find those that have been resolved and can be closed. Would you be able to close this issue or add a comment if there is something you don't think is resolved yet?

**Comment by MariaKokshaikina** [comment]
Hello @joelostblom
As you can see in the @jakevdp 's answer on StackOverflow, "The problem appears to be that in Vega, lookup transforms do not dynamically recompute in response to selections.", but there is a workaround provided. Not sure if the issue has to be closed (because it hasn't been resolved actually), but if there is no way to fix it in Altair right now, probably it can be put aside

**Comment by joelostblom** [comment]
Yes I believe this would have to be fixed in Vega-Lite / Vega, and I don't think there is anything that can be done about it in Altair. I suggest closing this and instead opening an issue in the Vega-Lite repo (unless you can find a similar one already open) and link it here.

**Comment by dangotbanned** [comment]
There seems to be a recent solution in https://stackoverflow.com/a/79281431

> I realize this is an older question, but for anyone still struggling to resolve the issue applying Jake's suggestion in their own code (as I did), I've found an alternative solution!

> I ended up ditching the "transform_lookup" altogether and handled it directly in GeoPandas instead:

    import geopandas as gpd
        
    url = data.world_110m.url #in my case, this was a github url
    gdf = gpd.read_file(url)
        
    merged_data = pd.merge(gdf, df, on='id', how='left')

> And then the code to the map would be something closer to a "normal" graph, just loading one data:


    alt.Chart(merged_data).mark_geoshape()\
        .encode(color='rate:Q')\
        .add_selection(select_date)\
        .transform_filter(select_date)\
        .project('equirectangular')\
        .properties(
            width=500,
            height=300,
            title='Title'
        )

> I hope this helps someone that is having difficulties making the transform_lookup work :)
